src/devices/SocketCan/Interop.cs

```diff
@@ -61,7 +67,7 @@ public static unsafe void Write(SafeHandle handle, ReadOnlySpan<byte> buffer)
                     int bytesWritten = Interop.SocketWrite((int)handle.DangerousGetHandle(), b, buffer.Length);
                     if (bytesWritten < 0)
                     {
-                        throw new IOException("`write` operation failed");
+                        throw new SocketException();
```

Comment:
can we instead create TryWrite to avoid throwing and catching? Also soon I believe we should be able to reuse socket class.

Even better option might be perhaps to use Socket class directly and remove most of the interop code since https://github.com/dotnet/runtime/issues/862 and https://github.com/dotnet/runtime/issues/29684 are now closed.

Reply:
(note https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.addressfamily?view=net-5.0 now has ControllerAreaNetwork value which we can use)

Reply:
I am aware of `ControllerAreaNetwork` value. If we want to use it, I can go with replacing Interop with the standard `Socket` instead of current *manual* approach.

Reply:
It would be perfect! I've been asking for these changes when I wrote this code. Please try not to remove any public surface when doing it though (new APIs are fine)

Reply:
Okay, I've started implementing :)

But the only issue I have encountered is that in order to use `AddressFamily.ControllerAreaNetwork`, we need to target *at least* `netcoreapp3.0`...

Reply:
I think we should eventually re-target as we can't be stuck with old version forever 😄 Before you proceed though let's double check with @joperezr @richlander @Ellerbach they don't have any objections to that so that your work doesn't go in vain 😃

Reply:
I'm open for retargeting. Rational:
- If anyone needs an update on an "older" version, there is always the source code, there is always git history
- it does allow to have access for advance bindings to more features like here and that's not blocking for the ones which should be able to get ported to nano and other smaller platforms
- it's like OS and tools, install updates :-) So move to the last version :-)

Reply:
We can either retarget to `netcoreapp3.1` **or** become `net5.0`-only.

(Fun fact, the last time I've checked .NET Core installation instructions was 3 days ago. 2 days ago .NET 5.0 has been released.)

Reply:
let's start with minimum you require to complete this. Note that projects for bindings can target different version than Iot.Device.Bindings, this way we can ensure that core nano bindings will still work